### PR TITLE
Compilation

### PR DESCRIPTION
**Pull request by rickbau5**
Implement compiling files to executable jar, downloading of that jar and importing/exporting of files as well.

@uidaho/lambda please review

### CODE REVIEW

**Comment by brandonjank** [comment]
@rickbau5 looks like travis checks failed on ccd33bf and 1dc7f0a

**Comment by rickbau5** [comment]
I noticed, having some strange errors that I've yet to track the source of down.
`PHP Fatal error:  Call to a member function delete() on null in /home/travis/build/uidaho/squireproject/tests/ProjectCommentTest.php on line 22`

**Comment by rickbau5** [comment]
@backburn alright that should do it

File: resources/views/editor/filelist.blade.php [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <table class="table table-striped table-hover">
     <thead>
         <tr>
-          <th>Filename</th>
-          <th>Type</th>
+          <th>Project Name</th>
```
Comment (brec9824) [inline resources/views/editor/filelist.blade.php]: Not really a big deal but do we really need the Project Name?

Comment (brandonjank) [inline resources/views/editor/filelist.blade.php]: nah, the first rows were for debugging. name, type, desc, creator, timestamps should be fine.

**Comment by rickbau5** [comment]
You can test compilation [here](https://boss2849.squireproject.com/editor/edit/Testing%20Project) using 'test_user' 'password'.

**Comment by brandonjank** [comment]
no buttons for compile/etc  on https://boss2849.squireproject.com/editor/edit/Testing%20Project/SquireFrame.java

**Comment by brandonjank** [comment]
iswydt, id recommend the compile stay on the editor page. its ok to have multiple ways to get to the same thing.

**Comment by brandonjank** [comment]
Looks like you still need to hand merge some stuff. 

From your project repository, bring in the changes and test.

```
git fetch origin
git checkout -b compile origin/compile
git merge master
```

**Comment by brec9824** [comment]
@rickbau5 Not really much time but I noticed two things. In the editor if you hit compile you dont get a success message and if you delete all files you get a error page saying Undefined offset: 0. But if you go back to the project page and back to the members page the main file has been readded so the error is gone. Neither of these are to big of a problem so let me know what you want to do because I am ready to merge this.

**Comment by rickbau5** [comment]
Yeah, so I have it set up so the Main file is automatically generated, even if it is deleted, however I guess that's only doing so on the project page and not the editor one. That's a quick fix

**Comment by rickbau5** [comment]
@brec9824 updated

File: resources/views/editor/filelist.blade.php [hunk 2]
```diff
@@ -3,7 +3,7 @@
         <tr>
           <th>File Name</th>
           <th>Description</th>
-          <th>Creator</th>
+          <th>Creator</th>wq
```
Comment (brec9824) [inline resources/views/editor/filelist.blade.php]: @rickbau5 I think this wq isnt supposed to be here

File: resources/views/editor/filelist.blade.php [hunk 3]
```diff
@@ -1,10 +1,9 @@
 <table class="table table-striped table-hover">
     <thead>
         <tr>
-          <th>Filename</th>
-          <th>Type</th>
+          <th>File Name</th>
           <th>Description</th>
-          <th>Creator</th>
+          <th>Creator</th>wq
```
Comment (brec9824) [inline resources/views/editor/filelist.blade.php]: @rickbau5 I think this wq isnt supposed to be here

**Comment by brec9824** [comment]
@rickbau5 The wq needs to be removed then I will merge

File: resources/views/editor/filelist.blade.php (hunk 3 above)
Comment (rickbau5) [inline resources/views/editor/filelist.blade.php]: yuup, had a moment in vim

**Comment by rickbau5** [comment]
done

**Comment by brec9824** [comment]
Merging

**Pull request merged**